### PR TITLE
Fix dashboard bugs

### PR DESCRIPTION
**Pull request by crusaderky**
- Fix bug in barcharts where upstream had an incorrect tooltip, showing only 1 run
- Fix bug in A/B charts which caused a crash in case of partially missing data
- Add optional dump of filtered pandas data, for debugging purposes

### CODE REVIEW

**Review by ian-r-rose — APPROVED** [review]

**Comment by ian-r-rose** [comment]
CI looks like it's stuck in a hanging state, even though tests have passed. Merging, thanks @crusaderky

File: dashboard.py [hunk 1]
```diff
@@ -242,12 +244,13 @@ def make_ab_confidence_map(
             "runtime",
         ]
     ]
+
     runtimes = df["runtime"].unique()
-    A = baseline
-    B = next((r for r in runtimes if r != baseline), None)
-    if not B:
```
Comment (ian-r-rose) [inline dashboard.py:248]: I still think this fix was necessary -- there are lots of ways to get subsets of data that only include the baseline numbers.

Comment (crusaderky) [reply dashboard.py:248]: On this line, you'll always get at most baseline + exactly one other runtime.
My change adds a check for the existence of baseline, but does not remove the check for the other runtime.

Comment (ian-r-rose) [reply dashboard.py:248]: If you have a subset of data with *only* baseline runtimes, your check still causes a `StopIteration` error.

Comment (crusaderky) [reply dashboard.py:248]: If you have only baseline, then the check 2 lines above `if len(runtimes) < 2` will fail.

**Pull request merged**